src/result/mod.rs

```diff
@@ -47,7 +48,13 @@ use std::fmt::{Display, Formatter, Result as FmtResult};
 ///
 /// [unit]: https://doc.rust-lang.org/book/ch05-01-defining-structs.html#unit-like-structs-without-any-fields
 #[derive(Debug)]
-pub struct Throw;
+pub struct Throw(PhantomData<*mut ()>); // *mut is !Send + !Sync, making it harder to forge
```

Comment:
Nit, `*mut` doesn't make it harder to forge, but it does make it harder to store.  It limits the UB to storing it in TLS.